addons/project/tests/test_project_flow.py

```diff
@@ -305,6 +306,19 @@ def test_task_with_no_project(self):
             except Exception as e:
                 raise AssertionError("Error raised unexpectedly while computing a field of the project ! Exception : " + e.args[0])
 
+        # tasks with no project set should only be visible to the users assigned to them
+        task_without_project.user_ids = [Command.link(self.user_projectuser.id)]
+        task_without_project.with_user(self.user_projectuser).read(['name'])
+        with self.assertRaises(AccessError):
+            task_without_project.with_user(self.user_projectmanager).read(['name'])
+
+        # Tests that tasks assigned to the current user should be in the 'inbox' stage by default
+        task = self.env['project.task'].create({
+            'name': 'Text Task!',
+            'user_ids': [Command.link(self.env.user.id)],
+        })
+        self.assertTrue(task.personal_stage_id.stage_id.name, 'Inbox')
```

Comment:
To confirm with @IT-Ideas , I think we could use the `_get_default_personal_stage_create_vals` method to get all default personal stages and we check the personal stage of the task is the element returned by `_get_default_personal_stage_create_vals` method.